ledger_entry and WebSocket Tool updates

mDuo13 opened the pull request:
Continues the work of #948, but with more extensive updates:

- Fixes some mistakes in `ledger_entry` description including the `type` field (not actually used in the code as far as I can tell!)
- Updates some examples in the WebSocket tool. In particular, discontinues the use of `r9cZA1mLK5R5Am25ArfXFmqgNwjZgnfk59` which probably belongs to a former Ripple employee of unknown identity
- Adds examples in the WebSocket tool for various `ledger_entry` modes

tjh3 committed:
TJ Harris Take-Home Test Option #1

Made several changes to structure of the page for clarity. Added extra request examples and split content to be more self-contained per call type. Some areas are structured correctly but I had to use Testnet instead of Mainnet.

tjh3 committed:
quick typo fix

Accidentally left text from earlier version of page

mDuo13 committed:
More ledger_entry and ws tool updates

tjh3 commented:
Agree with pretty much all the changes, except for removal of the (currently) optional `type` field. Made recommendation that field should be made required so that underlying method can check for that field first instead of inferring type of call based on first encountered field name (i.e. to avoid confusing "**Caution:** If you specify more than 1 of these type-specific fields in a request, the server retrieves results for only 1 of them. It is not defined which one the server chooses, so you should avoid doing this." message).

cjcobb23 commented:
> Agree with pretty much all the changes, except for removal of the (currently) optional `type` field. Made recommendation that field should be made required so that underlying method can check for that field first instead of inferring type of call based on first encountered field name (i.e. to avoid confusing "**Caution:** If you specify more than 1 of these type-specific fields in a request, the server retrieves results for only 1 of them. It is not defined which one the server chooses, so you should avoid doing this." message).

The `type` field is not required for `ledger_entry`. I often use `ledger_entry` without specifying a type, in situations where I don't know the type.

mDuo13 commented:
Like I said, from looking at the code, I wasn't able to find a part where the code even _looks at_ the `type` field in this method. (There are some other related methods that _do_ use a `type` field, formatted the same way, to good effect.)

mDuo13 committed:
ledger_entry - Mainnet examples for all types

- Working examples for all ledger_entry types in WebSocket tool
- Match examples on ledger_entry page to examples from WS tool
- Link various ledger_entry modes back & forth

ryangyoung commented on assets/js/apitool-methods-ws.js:
```diff
@@ -159,28 +165,148 @@ Request('ledger_data', {
 
-// Future feature: special case sign/sign_for so you can use those if you're
-// connected to a local server (with big warning)
+// Signing methods are not provided here because it's a terrible idea to send
```
trivial: Consider softening this a bit by briefly explaining _why_ it's a "terrible" idea (i.e., insecure, you could lose money, etc).

ryangyoung commented on content/references/rippled-api/public-rippled-methods/ledger-methods/ledger_entry.md:
```diff
@@ -41,144 +72,491 @@ An example of the request format:
+### Get AccountRoot Object
 
-The full list of parameters recognized by this method is as follows:
+Retrieve an [AccountRoot object](accountroot.html) by its address. This is roughly equivalent to the [account_info method][].
```
Is it worth enumerating any difference(s) between the two methods?

mDuo13 replied:
I don't really think so? The main difference is that account_info has some more options, as described in the doc, for fetching other stuff. There are some other inconsequential things, like `ledger_entry` puts its results in a `node` field rather than an `account_data` field (since it doesn't only return account data) and includes copies the ledger object ID to a higher-level field. Not really worth going into IMO.

ryangyoung commented on content/references/rippled-api/public-rippled-methods/ledger-methods/ledger_entry.md:
```diff
@@ -41,144 +72,491 @@ An example of the request format:
+{
+  "id": "example_get_escrow",
+  "command": "ledger_entry",
+  "escrow": {
```
Would it be worth providing a sample that shows `escrow` as a string as well?

mDuo13 replied:
Most or all of the alternate objects can be specified using their ledger object ID as a string. This is pretty much redundant with specifying the ledger object ID as a string in the `index` field instead, so I don't think it's worth doubling the number of examples to show.

mDuo13 committed:
Update why signing methods aren't in WS tool
